xhype/xhype/src/vmexit.rs

```diff
@@ -13,6 +16,105 @@ pub enum HandleResult {
     Next,
 }
 
+// Table 27-3
+#[inline]
+pub fn vmx_guest_reg(num: u64) -> X86Reg {
+    match num {
```

Comment:
On second thought....

This is basically `TryFrom` on a numeric type based on the qualification:

```
impl TryFrom<u32> for X86Reg {
    type Error = &'static str;

    fn try_from(src: u32) -> Result<Self> {
        match src {
            0 => Ok(Self::RAX),
            1 => Ok(Self::RCX),
            2 => Ok(Self::RDX),
            3 => Ok(Self::RBX),
            4 => Ok(Self::RSP),
            5 => Ok(Self::RBP),
            6 => Ok(Self::RSI),
            7 => Ok(Self::R7),
            8 => Ok(Self::R8),
            9 => Ok(Self::R9),
            10 => Ok(Self::R10),
            11 => Ok(Self::R11),
            12 => Ok(Self::R12),
            13 => Ok(Self::R13),
            14 => Ok(Self::R14),
            15 => Ok(Self::R15),
            _ => Err("bad register"),
        }
    }
}
```